charts/internal/cilium/values.yaml

```diff
@@ -332,24 +332,36 @@ global:
   encryption:
     # enabled enables encryption
     enabled: false
-
-    # keyFile is the name of the key file inside the Kubernetes secret
-    # configured via secretName
-    keyFile: keys
-
-    # mountPath is the path where to mount the secret inside the Cilium pod
-    mountPath: /etc/ipsec
-
-    # secretName is the name of the Kubernetes secret containing the encryption
-    # keys
-    secretName: cilium-ipsec-keys
+    # Encryption method. Can be either ipsec or wireguard.
+    mode: ipsec
```

Comment:
You're adding ipsec mode encryption as an option but I can't find a test for it. Please add an e2e test for the ipsec option or drop it and keep only wireguard.

Reply:
I will remove the IPsec mode from the `NetworkConfig` API. All of these helm chart things for IPsec were already in place, however never usable from the outside.
We can add the mode to the `NetworkConfig` later on